File: muxer-commons/src/main/java/pt/um/tf/commons/error/MissingExecutorException.java
package pt.um.tf.commons.error;

import io.atomix.catalyst.buffer.BufferInput;
import io.atomix.catalyst.buffer.BufferOutput;
import io.atomix.catalyst.serializer.CatalystSerializable;
import io.atomix.catalyst.serializer.Serializer;

public class MissingExecutorException extends Exception implements CatalystSerializable {
    @Override
    public String getMessage() {
        return "Missing Executor to execute task in.";
    }

    @Override
    public void writeObject(BufferOutput<?> buffer, Serializer serializer) {}

    @Override
    public void readObject(BufferInput<?> buffer, Serializer serializer) {}
}

File: muxer-commons/src/main/kotlin/pt/um/tf/taskmux/commons/error/MissingExecutorException.kt
package pt.um.tf.taskmux.commons.error

import io.atomix.catalyst.buffer.BufferInput
import io.atomix.catalyst.buffer.BufferOutput
import io.atomix.catalyst.serializer.CatalystSerializable
import io.atomix.catalyst.serializer.Serializer

class MissingExecutorException : Exception(), CatalystSerializable {
    override val message: String?
        get() = "Missing Executor to execute task in."


    override fun writeObject(buffer: BufferOutput<*>?, serializer: Serializer?) {}

    override fun readObject(buffer: BufferInput<*>?, serializer: Serializer?) {}

}
